warehouse/upload.go

```diff
@@ -1622,6 +1624,50 @@ func (job *UploadJobT) getLoadFilesTableMap() (loadFilesMap map[tableNameT]bool,
 	return
 }
 
+func (job *UploadJobT) destinationRevisionIDMap() (revisionIDMap map[string]backendconfig.DestinationT, err error) {
+	revisionIDMap = make(map[string]backendconfig.DestinationT)
+	revisionIDs, err := distinctDestinationRevisionIdsFromStagingFiles(struct {
+		sourceID           string
+		destinationID      string
+		startStagingFileID int64
+		endStagingFileID   int64
+	}{sourceID: job.warehouse.Source.ID, destinationID: job.warehouse.Destination.ID, startStagingFileID: job.upload.StartStagingFileID, endStagingFileID: job.upload.EndStagingFileID})
+	if err != nil {
+		return
+	}
+
+	var response []byte
+	var responseCode int
+
+	for _, revisionID := range revisionIDs {
+		// No need to make config backend api call for the current config
+		if revisionID == job.warehouse.Destination.RevisionID {
+			revisionIDMap[revisionID] = job.warehouse.Destination
+			continue
+		}
+
+		urlStr := fmt.Sprintf("%s/workspaces/destinationHistory/%s", configBackendURL, revisionID)
+
+		pkgLogger.Infof("[WH]: Get request Started for Dest revisionID %s", revisionID)
```

Comment:
Do we need these transient message logs for every revisionID ?

Reply:
Done.